aioredisqueue/queue.py

```diff
@@ -160,12 +170,18 @@ def _ack(self, task_id, method='multi'):
             args=[task_id],
         )
 
-    async def _requeue(self, before=None):
+    async def _requeue(self, now, before=-1):
         if 'requeue' not in self._lua_sha:
             await self._load_scripts('requeue')
 
         return await self._redis.evalsha(
             self._lua_sha['requeue'],
-            keys=[self._keys['ack'], self._keys['queue']],
-            args=[before],
+            keys=[self._keys['ack'], self._keys['queue'], self._keys['last_requeue']],
+            args=[now, self._requeue_interval, before],
         )
+
+    async def _requeue_periodically(self):
+        while not self._redis.closed:
```

Comment:
There should be a way to stop it without closing redis connection. Otherwise this method once started will prevent freeing resources.
Also a loop can be closed at this moment.